### PR TITLE
[WIP]Feature/fee

### PR DESCRIPTION
**Pull request by zakimal**
I tried to fix `circuit_transfer.rs` for fee feature, but my code failed to pass the test `test_circuit_transfer()` at `assert!(cs.is_satisfied());` (line 473) .

I want the `circuit` to pass a `zk-proof` calculated for the balance for which the `fee` and `transfer amount` have been collected.

I intend to implement `fee` in the same way as `value`, but something seems to be wrong. Please look at the code for advice. Thank you.

ref: 
> もう直接、クライアントでEnc(fee)作って、remaining_balance = balance - amont  - feeでzkして、Enc(fee), Txに乗っけて、オンチェーンでwithdrawするっていう方向性で開発いける？

### CODE REVIEW

**Comment by osuketh** [comment]
I'll check. Can you merge the master branch to trim my funking blanks?

File: core/proofs/src/circuit_transfer.rs [hunk 1]
```diff
@@ -342,62 +391,67 @@ fn u32_into_boolean_vec_le<E, CS>(
...
-        let address_sender_xy = proof_generation_key_s.into_encryption_key(params).0.into_xy(); 
-        let address_recipient_xy = address_recipient.0.into_xy();                       
-        
+        let address_sender_xy = proof_generation_key_s.into_encryption_key(params).0.into_xy();
+        let address_recipient_xy = address_recipient.0.into_xy();
+
         let alpha: fs::Fs = rng.gen();
 
         let value = 10 as u32;
         let remaining_balance = 17 as u32;
```
Comment (osuketh) [inline core/proofs/src/circuit_transfer.rs]: You need to fix here:
`let remaining_balance = 16 `

Comment (zakimal) [reply core/proofs/src/circuit_transfer.rs]: I overlooked that!

File: core/proofs/src/elgamal.rs [hunk 2]
```diff
@@ -192,7 +192,11 @@ mod tests {
         let params = &JubjubBls12::new();
         let p_g = FixedGenerators::NoteCommitmentRandomness; // 1
         let rng = &mut XorShiftRng::from_seed([0xbc4f6d44, 0xd62f276c, 0xb963afd0, 0x5455863d]);
+<<<<<<< HEAD
```
Comment (osuketh) [inline core/proofs/src/elgamal.rs]: conflicted

Comment (zakimal) [reply core/proofs/src/elgamal.rs]: yeah I modify it locally, so I'll push it.

**Comment by osuketh** [comment]
I just trim additional blanks in master branch

File: core/proofs/src/elgamal.rs (hunk 2 above)
Comment (zakimal) [reply core/proofs/src/elgamal.rs]: OMG my code pass the test!

File: core/proofs/src/circuit_transfer.rs (hunk 1 above)
Comment (zakimal) [reply core/proofs/src/circuit_transfer.rs]: OMG my code pass the test!

**Comment by MasashiSalvador57f** [comment]
reading this p-r (and diffs)

**Comment by osuketh** [comment]
This branch seems up-to-date.
https://github.com/LayerXcom/zero-chain/pull/109

**Pull request merged**